ABB.SrcML.Test/LibSrcMLRunnerTests.cs

```diff
@@ -0,0 +1,481 @@
+            }
+            catch (SrcMLException e) {
+                Console.WriteLine(e.Message);
+            }
+
+            {
+                Assert.True(File.Exists("output0.cpp.xml"));
+                SrcMLFile srcFile = new SrcMLFile("output0.cpp.xml");
+                Assert.IsNotNull(srcFile);
+
+                var files = srcFile.FileUnits.ToList();
+                Assert.AreEqual(1, files.Count());
+
+                string file = Path.Combine(TestInputPath, "input.cpp");
+                var f1 = (from ele in files
+                          where ele.Attribute("filename").Value == file
+                          select ele);
+                Assert.AreEqual(file, f1.FirstOrDefault().Attribute("filename").Value);
+            }
+            {
+                Assert.True(File.Exists("output1.cpp.xml"));
+                SrcMLFile srcFile = new SrcMLFile("output1.cpp.xml");
+                Assert.IsNotNull(srcFile);
+
+                var files = srcFile.FileUnits.ToList();
+                Assert.AreEqual(1, files.Count());
+
+                string file1 = Path.Combine(TestInputPath, "input2.cpp");
+                var f2 = (from ele in files
+                          where ele.Attribute("filename").Value == file1
+                          select ele);
+                Assert.AreEqual(file1, f2.FirstOrDefault().Attribute("filename").Value);
+            }
+            ad.Dispose();
+            bc.Dispose();
+        }
+
+        /// <summary>
+        /// Tests the creation of srcML from a file and returned via string
+        /// </summary>
+        [Test]
+        public void TestCreateSrcMLArchiveFtM() {
+            LibSrcMLRunner.SourceData ad = new LibSrcMLRunner.SourceData();
+            ad.SetArchiveFilename(Path.Combine(TestInputPath, "input.cpp"));
+            ad.SetArchiveLanguage(LibSrcMLRunner.SrcMLOptions.SRCML_LANGUAGE_CXX);
+
+            LibSrcMLRunner.SourceData bc = new LibSrcMLRunner.SourceData();
+            bc.SetArchiveFilename(Path.Combine(TestInputPath, "input2.cpp"));
+            bc.SetArchiveLanguage(LibSrcMLRunner.SrcMLOptions.SRCML_LANGUAGE_CXX);
+
+            List<LibSrcMLRunner.SourceData> data = new List<LibSrcMLRunner.SourceData>();
+            data.Add(ad);
+            data.Add(bc);
+
+            IntPtr structPtr = LibSrcMLRunner.CreatePtrFromStruct(ad);
+            IntPtr structPtr2 = LibSrcMLRunner.CreatePtrFromStruct(bc);
+
+            List<IntPtr> structArrayPtr = new List<IntPtr>();
+            structArrayPtr.Add(structPtr);
+            structArrayPtr.Add(structPtr2);
+            IntPtr s = new IntPtr(0);
+            try {
+                s = LibSrcMLRunner.SrcmlCreateArchiveFtM(structArrayPtr.ToArray(), structArrayPtr.Count());
+            }
+            catch (Exception e) {
+                Console.WriteLine("EXCEPTION: {0}",e.Message);
+                Assert.True(false);
```

Comment:
Is there a reason to catch/print the exception? Wouldn't it be more convenient to just let the exception get thrown so the testing framework catches it? That will cause the test to fail and whatever test runner people are using will very likely show the exception message.